resources/terraform/aws-eks/main.tf

```diff
@@ -0,0 +1,185 @@
+resource "random_string" "name_suffix" {
+  length  = 5
+  lower   = true
+  upper   = false
+  numeric = true
+  special = false
+}
+
+locals {
+  name         = "testenv-eks-${random_string.name_suffix.result}"
+  cluster_name = "${local.name}-cluster"
+}
+
+provider "aws" {
+  region = var.region
+}
+
+provider "kubernetes" {
+  host                   = module.eks.cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks.cluster_id]
+  }
+}
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+data "aws_eks_cluster_auth" "this" {
+  name = module.eks.cluster_id
+}
+
+module "vpc" {
+  source             = "terraform-aws-modules/vpc/aws"
+  version            = "~> 3.18"
+  enable_nat_gateway = true
+  cidr               = "10.0.0.0/16"
```

Comment:
This is not a big deal but worth pointing since we're trying to make these TF files consistent.

If you see your `module "eks"`, the `source` and `version` are always grouped together at the top and are separate from the other modules variables. I like this style and use it personally since it creates a visually separation between (a) what is the module/version and (b) what are the inputs.

I would suggest making all your modules follow this format.

```terraform
module "eks" {
  source  = "terraform-aws-modules/eks/aws"
  version = "~> 18.30"

  ...other variables
```